cmd/stdiscosrv/main.go

```diff
@@ -99,11 +106,15 @@ func main() {
 
 	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
 	if err != nil {
-		log.Println("Failed to load keypair. Generating one, this might take a while...")
-		cert, err = tlsutil.NewCertificate(certFile, keyFile, "stdiscosrv", 20*365)
-		if err != nil {
-			log.Fatalln("Failed to generate X509 key pair:", err)
+		log.Println("Failed to load keypair because:", err)
+		if !fileExists(certFile) && !fileExists(keyFile) {
```

Comment:
I'd go with just `if os.IsNotExist(err)` here; that should be sufficient

Reply:
for cert.pem or for key.pem or for both files?

Reply:
It's ok to just check the returned error from `LoadX509KeyPair`. If it's `os.IsNotExist` then we know that either the key or the cert didn't exist, and we can generate new ones.

Reply:
I fix that